### ISSUE TITLE
subscriptionId not sent when disposing subscription

### ISSUE THREAD

**Issue by jerelmiller** [absinthe-graphql/absinthe-socket#13]
Hello,

I am using @absinthe/socket with @absinthe/socket-relay packages and am having a hard time unsubscribing from the websocket.

I notice that when calling `dispose()` on the returned subscription from relay, it fails to send the `subscriptionId` to the phoenix. I see that the `subscriptionId` is coming back successfully upon connecting and that the `notifiers` array includes a notifier with a valid `subscriptionId`. I'm not sure if I am doing something wrong or not but any help would be appreciated.

Here is some relevant code for what I am trying to work with (backend is setup according to the absinthe subscription docs).

```javascript
// subscription
export default roomId =>
  requestSubscription(environment, {
    subscription: <query> 
    variables: { roomId }
  })


// relay config
const absintheSocket = AbsintheSocket.create(
  new Socket(`${process.env.REACT_APP_SOCKET_URL}/socket`)
)

const network = Network.create(fetchQuery, createSubscriber(absintheSocket))

// component
const MyComponent extends Component {
  componentDidMount() {
    this.subscription = mySubscription(<args>)
  }

  componentWillUnmount() {
    this.subscription.dispose()
  }

  render() {
    // ...
  }
}
```

I can see the reply coming back with the subscription ID

```
["1","2","__absinthe__:control","phx_reply",{"status":"ok","response":{"subscriptionId":"__absinthe__:doc:102122207"}}]
```

When disposing, no `subscriptionId` param is sent

```
["1","7","__absinthe__:control","unsubscribe",{}]
```

Let me know if you need any more info or have any guidance. Thanks!

**Comment by jerelmiller** [comment]
Seems to have fixed itself and I'm not entirely sure why. Maybe I had something misconfigured on the backend. I'll go ahead and close this unless I see something again.

**Comment by jerelmiller** [comment]
False alarm. Its still happening (so sorry).

**Comment by ssomnoremac** [comment]
hey @mgtitimoli any update on this? We are getting within a few weeks of shipping so I don't know how much longer we can hold off on being able to dispose subscriptions. Let me know if I can help in any way.

**Comment by davidye** [comment]
I ran into this as well, here's some more context:

When calling dispose(), the relay socket does not send a subscriptionId.

I tracked down the problem to this line:

https://github.com/absinthe-graphql/absinthe-socket/blob/05f7f2b9e8e7d5bcccbed07abf8fd8325db6891c/packages/socket-relay/src/createSubscriber.js#L24

When this dispose function is called, notifier.subscriptionId is still undefined.

Later on, onSubscriptionSucceed() is called and the notifier is updated with a valid subscriptionId.

https://github.com/absinthe-graphql/absinthe-socket/blob/950537a085ff57249954550ca0955b5156acc217/packages/socket/src/pushRequest.js#L28

However the notifier instance passed to cancel() is not the instance that was updated, and its subscriptionId is still undefined.
